django/apps/registry.py

```diff
@@ -329,5 +331,54 @@ def clear_cache(self):
             for model in self.get_models(include_auto_created=True):
                 model._meta._expire_cache()
 
+    def lazy_model_operation(self, function, *models_or_names):
+        """
+        Take a function and a number of models or model names, and when all
+        the corresponding models have been loaded, call the function with
+        the model classes as its arguments.
+
+        Each element of models_or names may be
+            * an actual model class
+            * a string of the form "app_label.ModelName"
+            * a tuple of strings of the form (app_label, model_name)
+
+        The function passed to this method must accept n positional arguments,
+        where n=len(models_or_names).
+        """
+
+        # Eagerly parse all model strings so we can fail immediately
+        # if any are plainly invalid.
+        from django.db.models.utils import make_model_tuple
```

Comment:
Do we need this import here or is it safe to put it at the top of the file? Maybe add a comment if it's a circular import if it needs to stay.